citylearn/electric_vehicle.py

```diff
@@ -0,0 +1,493 @@
+import inspect
+import math
+from typing import List, Mapping, Tuple, Union, Dict
+from gym import spaces
+import numpy as np
+from citylearn.base import Environment, EpisodeTracker
+from citylearn.data import EnergySimulation, CarbonIntensity, Pricing, Weather, EVSimulation
+from citylearn.energy_model import Battery, ElectricHeater, HeatPump, PV, StorageTank
+from citylearn.preprocessing import Normalize, PeriodicNormalization
+import random
+import copy
+
+class electric_vehicle(Environment):
+
+    def __init__(self, ev_simulation: EVSimulation,episode_tracker: EpisodeTracker, observation_metadata: Mapping[str, bool],
+                 action_metadata: Mapping[str, bool], battery: Battery = None, auxBattery: Battery = None, min_battery_soc: int = 20,
+                 image_path: str = None, name: str = None, **kwargs):
+        """
+        Initialize the EVCar class.
+
+        Parameters
+        ----------
+        ev_simulation : EVSimulation
+            Temporal features, locations, predicted SOCs and more.
+        battery : Battery
+            An instance of the Battery class.
+        observation_metadata : dict
+            Mapping of active and inactive observations.
+        action_metadata : dict
+            Mapping od active and inactive actions.
+        name : str, optional
+            Unique electric_vehicle name.
+
+        Other Parameters
+        ----------------
+        **kwargs : dict
+            Other keyword arguments used to initialize super class.
+        """
+
+        self.ev_simulation = ev_simulation
+        self.name = name
+
+        super().__init__(
+            seconds_per_time_step=kwargs.get('seconds_per_time_step'),
+            random_seed=kwargs.get('random_seed'),
+            episode_tracker=episode_tracker
+        )
+
+        self.battery = battery
+        self.aux_battery = auxBattery
+        self.observation_metadata = observation_metadata
+        self.action_metadata = action_metadata
+        self.non_periodic_normalized_observation_space_limits = None
+        self.periodic_normalized_observation_space_limits = None
+        self.observation_space = self.estimate_observation_space()
+        self.action_space = self.estimate_action_space()
+        self.min_battery_soc = min_battery_soc
+        self.image_path = image_path
+        self.__observation_epsilon = 0.0  # to avoid out of bound observations
+
+
+
+    @property
```

Comment:
Refer to comment in `citylearn.electric_vehicle_charger` about ordering of `property`, `setters` and how to write docstrings.